pkg/gitmanager/setup.go

```diff
@@ -174,3 +166,53 @@ func (gm *gitManager) initGit(ctx context.Context, payload *core.Payload, oauth
 	}
 	return nil
 }
+
+func (gm *gitManager) makeDownloadRequest(ctx context.Context, downloadURL string,
+	oauth *core.Oauth) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, downloadURL, http.NoBody)
+	if err != nil {
+		return nil, err
+	}
+	if oauth.AccessToken != "" {
+		req.Header.Add("Authorization", fmt.Sprintf("%s %s", oauth.Type, oauth.AccessToken))
+	}
+	resp, err := gm.httpClient.Do(req)
+	if err != nil {
+		gm.logger.Errorf("error while making http request %v", err)
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		gm.logger.Errorf("non 200 status while cloning from endpoint %s, status %d ", downloadURL, resp.StatusCode)
+		resp.Body.Close()
+		return nil, errs.ErrAPIStatus
+	}
+	return resp, nil
+}
+
+func (gm *gitManager) DownloadFileByCommit(ctx context.Context, gitProvider, repoSlug,
+	commitID, filePath string, oauth *core.Oauth) (string, error) {
+	downloadURL, err := urlmanager.GetFileDownloadURL(gitProvider, commitID, repoSlug, filePath)
+	if err != nil {
+		return "", err
+	}
+	resp, err := gm.makeDownloadRequest(ctx, downloadURL, oauth)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	path := utils.GenerateUUID() + ".yml"
```

Comment:
why a random path and not something like `<commit_id>/.tas.yml` in os.TempDir() folder? It'll cache the file for subsequent requests too. Also, use sync.Once to do this only once if file doesn't exist at the above path (See usage of `c.once.Do` in cachemanager).

Reply:
I wanted to clean the path (delete the file) after the processing it, as there is no merit in keeping the file (also it occupies some space) , keeping commit-id in path may lead to race condition in that case